experimental/dds/tree2/src/codec/parameterized-dds.md

```diff
@@ -0,0 +1,252 @@
+# Parameterized DDSes
+
+This design document centers around the problem of providing better support for "parameterized DDSes,"
+aka a DDS which is configurable in some way by an application author.
+The current APIs are generally sufficient for DDSes whose only parameters affect runtime behavior in a way that doesn't have compatibility issues.
+For example, one could imagine writing a DDS with a "noisy" mode that emits many more diagnostic events on changes but otherwise behaves the same.
+There are no compatibility concerns with this sort of parameter:
+clients within a collaboration session are free to disagree on whether the extra events are emitted,
+and a document created by a client with one level of verbosity can be later opened by a client with a different level.
+
+However, the APIs start to have issues when considering parameters which might result in compatibility issues.
+There are a few reasons why supporting such parameters is desirable:
+
+1. The DDS supports functionality requiring persisted data on which applications may want different policies, or may not want at all (and would like pay-to-play semantics)
+    - Current examples: attribution, legacy `SharedTree`'s `summarizeHistory` flag
+    - Potential other examples: DDS implementations with configurable merge behavior
```

Comment:
I'm not sure I agree with this example. I think right now we'd say that is a different DDS, and wouldn't want to be able to configure this level of behavior.

In general sense, I like to see features that leverage this to do no harm, specifically that any old clients continue to work. Data may be lost which is not good, but ideally it continues to function as if the new feature wasn't there. I think this is a better behavior for our consumers, as consumers likely need to make a choice on when to enable something, and generally 100% saturation is hard to do at scale, so the .0...9 percent should behave well enough. We should do our best to avoid that as well.

I mainly call this out, as i don't see this feature as being for upgrade, or shipping dark. We may need to use those techniques in the development of features that leverage this, but this feature is not meant for that. For instance, i would never expect the need to add FoobarReadOnlyShipDark before FoobarReal. They would just add FoobarReal which would gracefully handle data between itself, and other clients without it.